pkg/orchestration/wiring/wiringutil/compute/secret.go

```diff
@@ -0,0 +1,96 @@
+package compute
+
+import (
+	"strings"
+
+	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
+	"github.com/atlassian/voyager"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/knownshapes"
+	"github.com/pkg/errors"
+)
+
+type BindingResult struct {
+	ResourceName            voyager.ResourceName
+	BindableEnvVarShape     *knownshapes.BindableEnvironmentVariables
+	CreatedBindingFromShape smith_v1.Resource
+}
+
+func GenerateEnvVars(renameEnvVar map[string]string, bindingResults []BindingResult) ([]smith_v1.Reference, map[string]string, error) {
+	originalEnvVars := map[string]string{}
+	dependencyReferences := []smith_v1.Reference{}
+
+	for _, bindingResult := range bindingResults {
+		prefix := bindingResult.BindableEnvVarShape.Data.Prefix
+		bindingName := bindingResult.CreatedBindingFromShape.Name
+		resourceName := bindingResult.ResourceName
+
+		for envVarKey, path := range bindingResult.BindableEnvVarShape.Data.Vars {
+			smithReference := smith_v1.Reference{
+				Name:     wiringutil.ReferenceName(bindingName, strings.Split(path, ".")...),
+				Resource: bindingName,
+				Path:     path,
+				Modifier: smith_v1.ReferenceModifierBindSecret,
+			}
+
+			// Create the environment name {PREFIX}_{RESOURCE_NAME}_{KEY}
+			// Fail on any clashes
+			envVarName := strings.ToUpper(strings.Join([]string{prefix, string(resourceName), envVarKey}, "_"))
+			_, exists := originalEnvVars[envVarName]
+			if exists {
+				return nil, nil, errors.Errorf("clashing environment variable %q", envVarName)
```

Comment:
This is a divergence from current behavior (in micros1) - that's fine, but should be documented explicitly.